backend/routes/contactCreator.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import { type User } from '@prisma/client'
+import express from 'express'
+import prisma from '../utils/prismaClient'
+
+const contactCreatorRouter = express.Router()
+const db = prisma
+
+contactCreatorRouter.post('/', async function (req, res) {
+    try {
+        const contact = await db.contact.create({
+            data: {
+                firstName: req.body.firstName,
+                lastName: req.body.lastName,
+                email: req.body.email,
+                phoneNumber: req.body.phoneNumber,
+                notes: req.body.notes,
+                ownerID: (req.user as User).id,
+                source: req.body.source
+            }
+        })
+        for (const relationship of req.body.relationships) {
+            await db.userContactRelationship.create({
+                data: {
+                    relationshipType: relationship.type,
+                    contact: { connect: { id: contact.id } }
+                }
+            })
+        }
+        for (const importantDate of req.body.importantDates) {
+            await db.importantDateEvent.create({
+                data: {
+                    date: importantDate.date,
+                    eventType: importantDate.event,
+                    contact: { connect: { id: contact.id } }
+                }
+            })
```

Comment:
nit: we should be using `createMany`  instead of looping https://www.prisma.io/docs/concepts/components/prisma-client/crud#create-multiple-records